Makefile

```
@@ -2,6 +2,8 @@
 IMG ?= controller:latest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION = 1.31.0
+# The version of ETCD to run e2e tests against
+E2E_ETCD_VERSION ?= v3.5.21
```

We discussed this some weeks ago. But would it be better to default to the latest required rather than having this version hardcoded here? I'm afraid we may forget to update this version when we release new versions.

```suggestion
E2E_ETCD_VERSION ?= $(shell go list -m -f {{.Version}} go.etcd.io/etcd/api/v3)
```